pkg/api/types/volume_types.go

```diff
@@ -29,10 +57,22 @@ type VolumeLsCommandOptions struct {
 	Size bool
 	// Filter matches volumes based on given conditions
 	Filters []string
-	// containerd namespace
-	Namespace string
-	// Root directory of persistent nerdctl state
-	DataRoot string
-	// containerd address
-	Address string
+}
+
+type VolumePruneCommandOptions struct {
+	VolumeCommandOptions
+	// Client is the containerd client
+	Client *containerd.Client
+	// Context is underlying command context
+	Context context.Context
+}
+
+type VolumeRmCommandOptions struct {
+	VolumeCommandOptions
+	// Client is the containerd client
+	Client *containerd.Client
+	// Context is underlying command context
+	Context context.Context
```

Comment:
Putting the context into the struct should not be a good pattern. see discussion here https://github.com/containerd/nerdctl/pull/1792#discussion_r1060810151

Reply:
Is it preferable to put the `io.Writer` in the arguments rather than inside the struct? See https://github.com/containerd/nerdctl/pull/1792/files#diff-3d28f1d0455b7f054ebac7695bbbe3d9c758b44b12602f7365d404a86011a49eR45.

Reply:
> Is it preferable to put the `io.Writer` in the arguments rather than inside the struct? See https://github.com/containerd/nerdctl/pull/1792/files#diff-3d28f1d0455b7f054ebac7695bbbe3d9c758b44b12602f7365d404a86011a49eR45.

Yes, I think it better